Documentation for the ht16k33-char component

ilikecake opened the pull request:
## Description:

Adds documentation for the ht16k33-char display component

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** 

- esphome/esphome#8520

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/components/index.rst` when creating new documents for new components or cookbook.

ilikecake committed:
Initial commit

ilikecake committed:
Optimize the file size

ilikecake committed:
Create ht16k33-char.jpg

ilikecake committed:
Upload a better image

ilikecake committed:
Clarifications

ilikecake committed:
Trying to fix collapsable sections

ilikecake committed:
collapsible section fixes 2

ilikecake committed:
Fix spacing for nested lists

ilikecake committed:
Update ht16k33-char.rst

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Update ht16k33-char.rst

ilikecake committed:
Update ht16k33-char.rst

ilikecake committed:
Time period can be 0

ilikecake committed:
Add entry in index

ilikecake committed:
Merge branch 'next' into ht16k33

nagyrobi committed:
spell fix

ilikecake committed:
Add info on the supported Sparkfun device

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Merge branch 'next' into ht16k33

kbx81 committed:
Fix typos, remove duplicate config var (#5236)

jesserockz committed:
Merge branch 'current' into next

jesserockz committed:
Merge branch 'beta' into next

ssieb committed:
[psram] document disabled option (#5239)

Co-authored-by: Samuel Sieb <samuel@sieb.net>

jesserockz committed:
Merge branch 'beta' into next

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Adding and removing character codes.

ilikecake committed:
Add images for segment definitions

guillempages commented on components/display/ht16k33-char.rst:
```diff
@@ -0,0 +1,486 @@
+| 14-Segment Name | A  | B  | C  | D  | E  | F  | G1 | G2 | H  | J  | K  | L  | M  | N  | -- | -- |
++-----------------+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+
+If the character to add is already in the character map, its code will be overwritten by the provided code. Note that the bit number that corresponds to the semgments is the same regardless of the display type used. Internally the display may have a different mapping, but the python preprocessor rearranges the bits as needed.
```
```suggestion
If the character to add is already in the character map, its code will be overwritten by the provided code. Note that the bit number that corresponds to the segments is the same regardless of the display type used. Internally the display may have a different mapping, but the python preprocessor rearranges the bits as needed.
```

ilikecake committed:
Spelling

Co-authored-by: guillempages <guillempages@users.noreply.github.com>

ilikecake committed:
Resolve conflict with makefile

ilikecake committed:
Fix conflicts

ilikecake committed:
Update device IDs and buffer size terminology

ilikecake committed:
Try to fix conflict in index.rst

ilikecake committed:
Merge branch 'ht16k33' of https://github.com/ilikecake/esphome-docs into ht16k33

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Reorganize help files to new folder structure

ilikecake committed:
First pass at updating markdown formatting.

Updated documentation for HT16K33 character displays, including configuration examples and supported devices.

ilikecake committed:
Fix formatting

ilikecake committed:
Fix spacing of lists

Corrected formatting and consistency issues in the documentation for the HT16k33-char component, including optional parameters and supported characters.

ilikecake committed:
Remove extra spaces

ilikecake committed:
Fix numbering in device setup instructions

ilikecake committed:
Fix capitalization in markdown links and text

ilikecake committed:
Fix typo in ht16k33-char documentation

ilikecake committed:
Add image

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Move image to correct folder.

ilikecake committed:
Merge branch 'ht16k33' of https://github.com/ilikecake/esphome-docs into ht16k33

ilikecake committed:
Update image syntax for HT16K33 character display

ilikecake committed:
Fix image syntax in ht16k33-char documentation

Updated image syntax for better compatibility and removed caption from the first image.

ilikecake committed:
Remove extra space

ilikecake committed:
Merge branch 'ht16k33' of https://github.com/ilikecake/esphome-docs into ht16k33

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Fix yaml example.

ilikecake committed:
Try to fix conflict

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
I think this goes here.

ilikecake committed:
Merge branch 'ht16k33' of https://github.com/ilikecake/esphome-docs into ht16k33

ilikecake committed:
Fix formatting

ilikecake committed:
Move images

ilikecake committed:
Update formatting

ilikecake committed:
Fix formatting

ilikecake committed:
Fix formatting

ilikecake committed:
Fix formatting

ilikecake committed:
Update formatting

ilikecake committed:
Fix formatting

ilikecake committed:
Check image formatting

ilikecake committed:
more image formatting

ilikecake committed:
Remove old files

ilikecake committed:
Add a better picture

ilikecake committed:
Remove old images

ilikecake committed:
Identify features that are not implemented yet.

ilikecake committed:
Fix formatting

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Update ht16k33-char.mdx

ilikecake committed:
Update ht16k33-char.mdx

ilikecake committed:
Update ht16k33-char.mdx

ilikecake committed:
Update ht16k33-char.mdx

ilikecake committed:
Update ht16k33-char.mdx

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Merge branch 'next' into ht16k33

ilikecake committed:
Merge branch 'next' into ht16k33
